src/mcu_opcodes.cpp

```diff
@@ -342,6 +359,13 @@ void MCU_Jump_JMP(uint8_t operand)
             mcu.cp = (uint8_t)MCU_PopStack();
             mcu.pc = MCU_PopStack();
         }
+        else if (opcode_h == 0x19)
+        {
+            MCU_PushStack(mcu.pc);
+            MCU_PushStack(mcu.cp);
+            mcu.cp = mcu.r[opcode_l] & 0xff;
```

Comment:
odd case of opcode_l should be handled somehow, I guess hardware would just ignore LSB, so `opcode_l &= ~1` will work.